test-executor/src/main/java/com/cognifide/aet/executor/SuiteExecutor.java

```diff
@@ -0,0 +1,239 @@
+import com.cognifide.aet.rest.LockService;
+import com.cognifide.aet.rest.helpers.ReportConfigurationManager;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.felix.scr.annotations.Activate;
+import org.apache.felix.scr.annotations.Component;
+import org.apache.felix.scr.annotations.Property;
+import org.apache.felix.scr.annotations.Reference;
+import org.apache.felix.scr.annotations.Service;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.TimeUnit;
+
+import javax.jms.JMSException;
+import javax.jms.Session;
+
+/**
+ * This service is responsible for executing the test suite and maintaining the processing statuses.
+ * It creates the {@link SuiteRunner} and {@link SuiteStatusResult} queue for each test suite run
+ * and keeps those items in cache.
+ */
+@Service(SuiteExecutor.class)
+@Component(label = "AET Suite Executor", description = "Executes received test suite", immediate = true,
+    metatype = true)
+public class SuiteExecutor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SuiteExecutor.class);
+
+  private static final String RUNNER_IN_QUEUE = "AET.runner-in";
+
+  private static final String HTML_REPORT_URL_FORMAT = "%s/report.html?company=%s&project=%s&correlationId=%s";
+
+  private static final String XUNIT_REPORT_URL_FORMAT = "%s/xunit?company=%s&project=%s&correlationId=%s";
+
+  private static final String MESSAGE_RECEIVE_TIMEOUT_PROPERTY_NAME = "messageReceiveTimeout";
+
+  private static final long DEFAULT_MESSAGE_RECEIVE_TIMEOUT = 20000L;
+
+  @Property(name = MESSAGE_RECEIVE_TIMEOUT_PROPERTY_NAME, label = "ActiveMQ message receive timeout",
+      description = "ActiveMQ message receive timeout", longValue = DEFAULT_MESSAGE_RECEIVE_TIMEOUT)
+  private Long messageReceiveTimeout;
+
+  private static final String CACHE_EXPIRATION_TIMEOUT_PROPERTY_NAME = "cacheExpirationTimeout";
+
+  private static final long DEFAULT_CACHE_EXPIRATION_TIMEOUT = 30000L;
+
+  @Property(name = CACHE_EXPIRATION_TIMEOUT_PROPERTY_NAME, label = "Suite runner cache expiration timeout",
+      description = "Suite runner cache expiration timeout. Should be longer than 'ActiveMQ message receive timeout'.",
+      longValue = DEFAULT_CACHE_EXPIRATION_TIMEOUT)
+  private Long cacheExpirationTimeout;
+
+  @Reference
+  private JmsConnection jmsConnection;
+
+  @Reference
+  private LockService lockService;
+
+  @Reference
+  private ReportConfigurationManager reportConfigurationManager;
+
+  private Cache<String, SuiteRunner> suiteRunnerCache;
+
+  private Cache<String, Queue<SuiteStatusResult>> suiteStatusCache;
+
+  private RunnerCacheUpdater runnerCacheUpdater;
+
+  private SuiteStatusHandler suiteStatusHandler;
+
+  @Activate
+  public void activate(Map properties) {
```

Comment:
Would we ever use a non-string key? How about using `Map<String, Object>` instead?